handler_test.go

```diff
@@ -0,0 +1,1982 @@
+				},
+			},
+		},
+		expectedStatuses: []ocpnetworkapiv1alpha1.DNSNameResolverStatus{
+			{
+				ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+					{
+						DNSName: "*.example.com.",
+						ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+							{
+								IP:         "1.1.1.1",
+								TTLSeconds: 30,
+							},
+							{
+								IP:         "1.1.1.2",
+								TTLSeconds: 30,
+							},
+						},
+						ResolutionFailures: 0,
+						Conditions: []metav1.Condition{
+							{
+								Type:    ConditionDegraded,
+								Status:  metav1.ConditionFalse,
+								Reason:  dns.RcodeToString[dns.RcodeSuccess],
+								Message: rcodeMessage[dns.RcodeSuccess],
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		name: "Remove regular dns names from wildcard dns name resolver object status and add the wildcard dns name",
+		dnsNameResolvers: []ocpnetworkapiv1alpha1.DNSNameResolver{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wildcard",
+					Namespace: "dns",
+				},
+				Spec: ocpnetworkapiv1alpha1.DNSNameResolverSpec{
+					Name: "*.example.com.",
+				},
+				Status: ocpnetworkapiv1alpha1.DNSNameResolverStatus{
+					ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+						{
+							DNSName: "www.example.com.",
+							ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+								{
+									IP:         "1.1.1.1",
+									TTLSeconds: 30,
+								},
+								{
+									IP:         "1.1.1.2",
+									TTLSeconds: 30,
+								},
+							},
+							ResolutionFailures: 0,
+							Conditions: []metav1.Condition{
+								{
+									Type:    ConditionDegraded,
+									Status:  metav1.ConditionFalse,
+									Reason:  dns.RcodeToString[dns.RcodeSuccess],
+									Message: rcodeMessage[dns.RcodeSuccess],
+								},
+							},
+						},
+						{
+							DNSName: "sub.example.com.",
+							ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+								{
+									IP:         "1.1.1.1",
+									TTLSeconds: 30,
+								},
+								{
+									IP:         "1.1.1.2",
+									TTLSeconds: 30,
+								},
+							},
+							ResolutionFailures: 0,
+							Conditions: []metav1.Condition{
+								{
+									Type:    ConditionDegraded,
+									Status:  metav1.ConditionFalse,
+									Reason:  dns.RcodeToString[dns.RcodeSuccess],
+									Message: rcodeMessage[dns.RcodeSuccess],
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		dnsQueryTestCases: []test.Case{
+			{
+				Qname: "*.example.com.",
+				Qtype: dns.TypeA,
+				Rcode: dns.RcodeSuccess,
+				Answer: []dns.RR{
+					test.A("*.example.com. 30 IN A 1.1.1.1"),
+					test.A("*.example.com. 30 IN A 1.1.1.2"),
+				},
+			},
+		},
+		expectedStatuses: []ocpnetworkapiv1alpha1.DNSNameResolverStatus{
+			{
+				ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+					{
+						DNSName: "*.example.com.",
```

Comment:
For my understanding. Why the wildcard supersedes all the subdomain resolved names? Is this only for the case when the IPs match the wildcard ones?

Reply:
Explained here: https://github.com/openshift/coredns-ocp-dnsnameresolver/pull/1#discussion_r1526232810



> Is this only for the case when the IPs match the wildcard ones?

Yes. If the IP addresses don't match the resolved name of the regular DNS name with a different set of IP addresses will be maintained.